stdlib/public/Cxx/std/String.swift

```diff
@@ -95,6 +95,30 @@ extension std.u32string {
   }
 }
 
+extension std.wstring {
+  /// Creates a C++ wide character string having the same content as the given 
+  /// Swift string.
+  ///
+  /// Note that the definition of a wide character differs across platforms:
+  /// it is UTF-16 on Windows but UTF-32 on other platforms.
+  ///
+  /// - Complexity: O(*n*), where *n* is the number of wide characters in the
+  ///   Swift string.
+  @_alwaysEmitIntoClient
+  public init(_ string: String) {
+    self.init()
+#if os(Windows)
+    for char in string.utf16 {
```

Comment:
Would relying on something like `string.withCString(encodedAs: UTF16.self)` be more efficient?

Reply:
Quite possibly, yes, although this is copied verbatim from the `std::u16string` constructor, so I think we should update both constructors in a separate PR.